.github/workflows/comment.yaml

```diff
@@ -13,57 +13,44 @@ env:
   ARTIFACT_NAME: "report"
 
 jobs:
-  download-artifacts:
-    name: Download Artifacts
+  filter-artifacts:
+    name: Filter Artifacts
     runs-on: ubuntu-24.04
     if: github.event.workflow_run.event == 'pull_request'
     outputs:
-      artifacts: ${{ fromJson(steps.download-artifact.outputs.result) }}
+      artifacts: ${{ fromJson(steps.filter-artifacts.outputs.result) }}
     steps:
       - uses: actions/checkout@v4.2.2
-      - id:   download-artifact
-        name: Download test report artifact
+      - id:   filter-artifacts
+        name: Filter test report artifacts
         uses: actions/github-script@v7.0.1
         with:
           script: |
             const fs = require('fs');
-            const owner = context.payload.workflow_run.repository.owner.login;
-            const repo = context.payload.workflow_run.repository.name;
-            const run_id = ${{ github.event.workflow_run.id }};
             const artifacts = await github.rest.actions.listWorkflowRunArtifacts({
-                owner: owner,
-                repo: repo,
-                run_id: run_id,
+                owner: context.payload.workflow_run.repository.owner.login,
+                repo: context.payload.workflow_run.repository.name,
+                run_id: ${{ github.event.workflow_run.id }},
             });
-            const report_artifacts = artifacts.data.artifacts.filter(artifact =>
+            const zips = artifacts.data.artifacts.filter(artifact =>
               artifact.name.startsWith('${{ env.ARTIFACT_NAME }}-')
-            )
-            const zips = await Promise.all(
-              report_artifacts.map(async artifact => {
-                const download = await github.rest.actions.downloadArtifact({
-                    owner: owner,
-                    repo: repo,
-                    artifact_id: artifact.id,
-                    archive_format: 'zip',
-                });
-                const zipPath = `${artifact}.zip`;
-                fs.writeFileSync(zipPath, Buffer.from(download.data));
-                return zipPath;
-              })
             );
             return JSON.stringify(zips);
   comment-on-prs:
     name: Comment on PRs
     runs-on: ubuntu-24.04
-    needs: download-artifacts
+    needs: filter-artifacts
     if: github.event.workflow_run.event == 'pull_request'
     strategy:
       matrix:
-        artifact: ${{fromJson(needs.download-artifacts.outputs.artifacts)}}
+        artifact: ${{fromJson(needs.filter-artifacts.outputs.artifacts)}}
     steps:
-      - name: Unzip
-        shell: bash
-        run: unzip ${{ matrix.artifact }}
+      - name: Download artifact
+        uses: actions/download-artifact@v4
```

Comment:
Can you pin this to a specific version so that builds are closer to reproducible?

Reply:
Done